admin/controllers/pupil.js

```diff
@@ -10,81 +12,64 @@ const pupilService = require('../services/pupil.service')
 const pupilDataService = require('../services/data-access/pupil.data.service')
 const azureFileDataService = require('../services/data-access/azure-file.data.service')
 const pupilUploadService = require('../services/pupil-upload.service')
+const schoolDataService = require('../services/data-access/school.data.service')
+const pupilAddService = require('../services/pupil-add-service')
+
+const controller = {}
 
-const getAddPupil = async (req, res, next) => {
+controller.getAddPupil = async (req, res, next, error = null) => {
```

Comment:
Are we changing the naming convention? This will require some refactoring of existing controllers.

Reply:
It's still the same name.  It was necessary to change the the exports for unit testing - as it was it was impossible to stub the getAddPupil from the postAddPupil.

I'd suggest this is a more testable pattern.

Reply:
The call signature was updated to allow the error page to displayed, as well as the default initial page.